kpi/tests/api/v2/test_api_submissions.py

```diff
@@ -301,30 +327,43 @@ def test_list_submissions_with_partial_permissions(self):
         partial_perms = {
             PERM_VIEW_SUBMISSIONS: [{'_submitted_by': self.someuser.username}]
         }
-        response = self.client.get(self.submission_url, {"format": "json"})
+        response = self.client.get(self.submission_list_url, {"format": "json"})
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
         self.asset.assign_perm(self.anotheruser, PERM_PARTIAL_SUBMISSIONS,
                                partial_perms=partial_perms)
-        response = self.client.get(self.submission_url, {"format": "json"})
+        response = self.client.get(self.submission_list_url, {"format": "json"})
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertTrue(self.asset.deployment.submission_count == 2)
+
+        # Calculate number of submissions that matches partial permissions
+        # condition.
+        all_submissions = self._deployment.get_submissions(self.asset.owner)
+        viewable_submissions_count = 0
+        for submission in all_submissions:
+            if {'_submitted_by': submission['_submitted_by']} in partial_perms[
+                PERM_VIEW_SUBMISSIONS
+            ]:
+                viewable_submissions_count += 1
+
         # User `anotheruser` should only see submissions where `submitted_by`
         # is filled up and equals to `someuser`
-        self.assertTrue(response.data.get('count') == 1)
-        submission = response.data.get('results')[0]
-        self.assertTrue(submission.get('_submitted_by') == self.someuser.username)
+        self.assertTrue(response.data.get('count') == viewable_submissions_count)
+        for submission in response.data['results']:
+            self.assertTrue(
+                {'_submitted_by': submission['_submitted_by']}
+                in partial_perms[PERM_VIEW_SUBMISSIONS]
```

Comment:
what's the advantage of looking this up in `PERM_VIEW_SUBMISSIONS` over writing `self.someuser.username`?

Reply:
![image](https://media1.giphy.com/media/26FL6V17him5yRF8k/200.gif)